go.mod

```diff
@@ -1,13 +1,23 @@
 module github.com/caddyserver/certmagic
 
-go 1.14
+go 1.17
```

Comment:
I don't think we can bump to 1.17 just yet, Fedora/RHEL don't ship with Go 1.17 yet. We might be able to bump it up in mid-May though.

Reply:
`io/fs` was new in 1.16, so we should at least bump to 1.16

Reply:
(though personally I'd advocate for going to 1.17 - I think being held back by RedHat is pointless 😝)

Reply:
Does this line actually require anyone importing it to have that Go version or newer though?

In any case, I'm not super inclined to delay just a month or two for RedHat's benefit.

Reply:
I'm pretty sure it does, yes.

Reply:
Okay, but the implication is that we still can't use new features in this lib until Caddy is also ready to bump, else it won't compile.

I'm just not sure what the benefit is to marking the `go.mod` as 1.17 if we're not actually using 1.17 features here yet.

Reply:
There are Go toolchain features offered by Go 1.17 that we can start using